docs/modules/operation/pages/deep-dive/alarms/alarm-handling.adoc

```diff
@@ -0,0 +1,51 @@
+
+== Alarm handling
+
+The following interactions with alarms are possible.
+
+=== Acknowledgment
+
+PICTURES MISSING
+
+An alarm can be acknowledged by an user.
+The alarm will be moved out from `Alarm(s) outstanding` into `Alarm(s) acknowledged` view.
+By doing this other OpenNMS users can notice, that someone is aware of this alarm.
+
+NOTE: A manually acknowledged alarm that was cleared automatically by its clear event needs also unacknowledged manually to disappear.
```

Comment:
I'm not clear what you mean here. 
Alarm was manually acknowledged. Clear event cleared it. They then need to manually unacknowledge it? Is that because it remains visible as long as it is marked acknowledged?

Reply:
> Alarm was manually acknowledged. Clear event cleared it. They then need to manually unacknowledge it?

Yes. That is the reality. You need to unack it to let it disappear from the alarm list.

> Is that because it remains visible as long as it is marked acknowledged?

That is also a good explanations. Acked alerts remain visible even if a clear event cleared it.

Reply:
This isn't true.  _All_ alarms will eventually be removed automatically.  It's just that ack'ed alarms are kept longer than un-acked alarms.

Reply:
Per `alarmd.drl`, un-acked alarms are kept for 5 minutes after clearing.  Acked alarms are kept for 1 day after clearing.  This is customizable by the user editing the cleanup rules in the alarmd drools.

Reply:
I am pretty sure I saw this behavior often in my system. And I didn't change drools files for sure.
But let me add @mmahacek 's point into the docs, since they reflect how it should work. When I see it in the real world again, I can open a bug report ;-)

Reply:
Default alarmd cleanup interval below.  This is based on the last event time, so any new event coming in and getting reduced into an existing alarm will reset this timer.  We try to purge alarms from the database as any open alarm requires overhead in drools to keep track of it.  Not much of an issue on smaller instances, but can become more resource intensive when there's a high number of active alarms.

|Criteria|Cleanup Interval|
|---|---|
|Cleared and Un-ack|5 mins|
|Cleared and Acked|1 day|
|All Un-ack|3 days|
|All alarms|8 days|